prisma/schema.prisma

```diff
@@ -23,6 +23,73 @@ datasource db {
+model achievement_template {
+  id                            Int                       @id @default(autoincrement())
+  metrics                       String[]                  
+  templateFor                   achievement_template_for_enum
+  // Groups defines the "final-achievement"
+  group                         String                    @db.VarChar
+  // Group order tells in which place a subordinate step / achievement is
+  groupOrder                    Int                      
+  // the final achievement is marked with isLast 
+  isLast                        Boolean                   
+  // what type of achievement is it
+  type                          achievement_type_enum
+  name                          String                    @db.VarChar
+  subtitle                      String                    @db.VarChar
+  description                   String                    @db.VarChar
+  image                         String                    @db.VarChar
+  // some achievements show actions that a user can/must perform next 
+  actionDescription             String?                   @db.VarChar
+  actionRedirectLink            String?                   @db.VarChar
+  actionType                    achievement_action_type_enum?
+  // defines what value must be reached to receive the achievement, e.g. "duration == 60", "count == 5", "count == 3 & duration == 60", "series >= 5" (https://github.com/onlabsorg/swan-js)
+  condition                     String                    @db.VarChar
+  // defined how values must be aggregated to evaluate the condition
+  conditionDataAggregations     Json                      @db.Json
+  isActive                      Boolean 
+}
+model user_achievement {
+  id                            Int                       @id @default(autoincrement())
+  metrics                       String[]                  @default([])
+  templateFor                   achievement_template_for_enum
+  group                         String                    @unique @db.VarChar
+  groupOrder                    Int                       
+  isLast                        Boolean                   
+  type                          achievement_type_enum
+  name                          String                    @db.VarChar
+  subtitle                      String                    @db.VarChar
+  description                   String                    @db.VarChar
+  image                         String                    @db.VarChar
+  actionDescription             String?                   @db.VarChar
+  actionRedirectLink            String?                   @db.VarChar
+  actionType                    achievement_action_type_enum?         
+  condition                     String                    @db.VarChar
+  conditionDataAggregations     Json                      @db.Json
+  isActive                      Boolean                   @default(true) 
+  isSeen                        Boolean                   @default(false)
```

Comment:
Do we need this per achievement? Couldn't we do it like we do with notifications (achievement.updatedAt > user.lastCheckedAchievements)?

Reply:
Or what does this boolean do, maybe I misunderstand

Reply:
We also thought of this, but there is a problem. As a user you could have seen the achievement on many different pages: In the course detail view, on you match detail view or on the achievements overview. This makes it hard to keep track of it with one `lastCheckedAchievements`, because you would either mark some achievements as seen that haven't really been seen or you have maintain like 3 different timestamps.

Reply:
I see makes sense